### PR TITLE
ut changes for exercising getmax and getmin for btree

### PR DESCRIPTION
**Pull request by shashank-parulekar**

### CODE REVIEW

File: btree/btree.c [hunk 1]
```diff
@@ -6888,7 +6889,48 @@ struct btree_ut_thread_info {
...
+
+#define  UT_START_THREADS()                                                   \
+	thread_run = true
+
+#define  UT_STOP_THREADS()                                                    \
+	thread_run = false
+
+#define UT_THREAD_WAIT()                                                      \
+	do {                                                                  \
+		while(!thread_run)                                            \
```
Comment (NikhilBirgade) [inline btree/btree.c]: while (

File: btree/btree.c [hunk 2]
```diff
@@ -6888,7 +6889,48 @@ struct btree_ut_thread_info {
...
+		}                                                             \
+	} while (0)
+
+#define UT_REQUEST_PEER_THREADS_TO_QUIESCE()                                  \
+	do {                                                                  \
+		bool try_again;                                               \
+		do {                                                          \
+			try_again = false;                                    \
+			if (m0_atomic64_cas(&thread_run, true, false)) {      \
+				while(m0_atomic64_get(&threads_quiesced) <    \
```
Comment (NikhilBirgade) [inline btree/btree.c]: while (

File: btree/btree.c [hunk 3]
```diff
@@ -6888,7 +6889,48 @@ struct btree_ut_thread_info {
...
+			UT_THREAD_WAIT();                                     \
+			m0_atomic64_dec(&threads_quiesced);                   \
+		}                                                             \
+	} while (0)
+
+#define UT_REQUEST_PEER_THREADS_TO_QUIESCE()                                  \
+	do {                                                                  \
+		bool try_again;                                               \
+		do {                                                          \
+			try_again = false;                                    \
```
Comment (NikhilBirgade) [inline btree/btree.c:6923]: `try_again` can be initialised as `false` on line 6921.

File: btree/btree.c [hunk 4]
```diff
@@ -6888,7 +6889,48 @@ struct btree_ut_thread_info {
  *  for the same btree nodes to work on thus exercising possible race
  *  conditions.
  */
-static volatile bool thread_start = false;
+static int64_t thread_run;
```
Comment (NikhilBirgade) [inline btree/btree.c:6892]: data type should be `bool`

File: btree/btree.c (hunk 3 above)
Comment (shashank-parulekar) [reply btree/btree.c:6923]: No this will not work. We need to re-initialize it in the loop.

File: btree/btree.c (hunk 4 above)
Comment (shashank-parulekar) [reply btree/btree.c:6892]: unfortunately I cannot use bool because the atomic routine needs int64_t type.

File: btree/btree.c (hunk 2 above)
Comment (shashank-parulekar) [reply btree/btree.c]: done

File: btree/btree.c (hunk 1 above)
Comment (shashank-parulekar) [reply btree/btree.c]: done

File: btree/btree.c (hunk 4 above)
Comment (NikhilBirgade) [reply btree/btree.c:6892]: Okay.

**Pull request merged**